src/transformers/pipelines/feature_extraction.py

```diff
@@ -79,4 +79,10 @@ def __call__(self, *args, **kwargs):
         Return:
             A nested list of :obj:`float`: The features computed by the model.
         """
-        return super().__call__(*args, **kwargs).tolist()
+        results = super().__call__(*args, **kwargs)
+        if hasattr(results, "tolist"):
+            results = results.tolist()
+        else:
```

Comment:
Test for a list/tuple and then raise a `ValueError` if we are not in that case? Seems a bit cleaner.

Reply:
No it's linked to sequential vs batched, so both are valid.

I will invert both cases to avoid `hasattr` but I want to avoid code specific to the type of tensor that's the reason it's written this way.